[DOC] Update the Pyroscope data source config doc (#89648)
grafana/grafana: docs/sources/datasources/pyroscope/configure-pyroscope-data-source.md

1. Click **Connections** in the left-side menu.
1. Under Your connections, click **Data sources**.
1. Enter `Grafana Pyroscope` in the search bar.
1. Select **Add new data source**.
1. Click **Grafana Pyroscope** to display the **Settings** tab of the data source.
1. Set the data source's basic configuration options.
1. Select **Save & test**.

## Configuration options

You can configure several options for the Pyroscope data source, including the name, HTTP, authentication, querying, and private data source connect.

If you make any changes, select **Save & test** to preserve those changes.

![Configuration options for the Pyroscope data source](/media/docs/grafana/data-sources/screenshot-pyroscope-data-source-config.png)

### Name and default

**Name**
: Enter a name to specify the data source in panels, queries, and Explore.

**Default**

> : The default data source will be pre-selected for new panels.

Reviewer: Maybe?
```suggestion
: The default data source is pre-selected for new panels.
```

### HTTP

The HTTP section is shown in number 1 in the screenshot.

**URL**
: The URL of the Grafana Pyroscope instance, for example, `https://localhost:4100`.

**Allowed cookies**
: The Grafana Proxy deletes forwarded cookies. Use this field to specify cookies by name that should be forwarded to the data source.

**Timeout**
: HTTP request timeout in seconds.

> ### Auth

Reviewer: There's a mixture of casing in the description list but I'm gonna assume that matches the UI

Author: Yes. It's one of those things we can work on later.

Author: Okay, so the data source screen has changed since I started this PR. 
![image](https://github.com/grafana/grafana/assets/104772500/9b0b2ceb-9bfd-4587-83c9-21149ee26516)

I need to update the instructions and the titles.

Author: This was the old screen 
![image](https://github.com/grafana/grafana/assets/104772500/aa708e5c-8605-4422-839d-82ceb949ae01)

Author: Okay, so the data source was updated and will be doing out with the next release. It is probably useful to merge these changes for 11.1 and then update the data source docs for the next release.

The Auth section is shown in number 2 in the screenshot.

**Basic auth**
: Enable basic authentication to the data source. When activated, it provides **User** and **Password** fields.

**With Credentials**
: Whether credentials, such as cookies or auth headers, should be sent with cross-site requests.

**TLS Client Auth**

> : Toggle on to use client authentication. When enabled, it adds the `Server name`, `Client cert` and `Client key`. The client provides a certificate that is validated by the server to establish the client's trusted identity. The client key encrypts the data between client and server. These details are encrypted and stored in the Grafana database.

Reviewer: Are these fields like with **Basic auth** above? If so, should it be:

```suggestion
: Toggle on to use client authentication. When enabled, it adds the **Server name**, **Client cert**, and **Client key** fields. The client provides a certificate that is validated by the server to establish the client's trusted identity. The client key encrypts the data between client and server. These details are encrypted and stored in the Grafana database.
```

**With CA Cert**
: Activate this option to verify self-signed TLS certificates.

**Skip TLS Verify**
: When activated, it bypasses TLS certificate verification.

**Forward OAuth Identity**

> : When activated, the user’s upstream OAuth identity is forwarded to the data source along with their access token.

Reviewer: https://github.com/grafana/writers-toolkit/blob/main/vale/Grafana/OAuth.yml

```suggestion
: When activated, the user’s upstream OAuth 2.0 identity is forwarded to the data source along with their access token.
```

**Custom HTTP Headers**
: Select Add header to add Header and Value fields.

**Header**
: Add a custom header. This allows custom headers to be passed based on the needs of your Pyroscope instance.

**Value**
: The value of the header.

### Querying

The **Querying** section is shown in number 3 in the screenshot.

> **Minimum step** is used for queries returning timeseries data. The default value is 15 seconds.

Reviewer: https://grafana.com/docs/writers-toolkit/write/style-guide/word-list/#time-series

```suggestion
**Minimum step** is used for queries returning time-series data. The default value is 15 seconds.
```

> The Pyroscope backend, similar to Prometheus, scrapes profiles at certain intervals. To prevent querying at smaller interval, use Minimal step same or higher than your Pyroscope scrape interval. This prevents returning too many data points to the frontend.

Reviewer: ```suggestion
The Pyroscope backend, similar to Prometheus, scrapes profiles at certain intervals. To prevent querying at smaller interval, use a minimum step that is the same or higher than your Pyroscope scrape interval. This prevents returning too many data points to the frontend.
```

> The Pyroscope backend, similar to Prometheus, scrapes profiles at certain intervals. To prevent querying at smaller interval, use Minimal step same or higher than your Pyroscope scrape interval. This prevents returning too many data points to the frontend.

Reviewer 2: The equivalent setting in Prometheus is https://github.com/grafana/grafana/blame/a16782a2d6c5823dd8d1abdc463da9f8d8b8113a/docs/sources/datasources/prometheus/configure-prometheus-data-source.md#L112. Unfortunately the name of the setting and therefore the  doc page is not very consistent with how it is done there.

In general the scrape interval is not configured in the Pyroscope backend and rather either in the SDKs or in Grafana Agent or Alloy. It basically is the value how often profiles would be typically collected. When comparing with grafana alloy it is called the "scrape_interval" there.

This is therefore a wrong statement: `The Pyroscope backend, similar to Prometheus, scrapes profiles at certain intervals.` as Pyroscope never scrapes.

`This prevents returning too many data points to the frontend`, that might be true, but the actual reason for this parameters is to prevent gaps when you zoom in too much. (As reported here: https://github.com/grafana/explore-profiles-deprecated/issues/399).

Author: Thank you for the clear explanation. I think I pulled that explanation from the Prometheus docs, and wasn't sure if ti was correct or not. In the previous version of the data source screen (used for this doc), the additional information in the hover text for Minimum step said: 
![image](https://github.com/grafana/grafana/assets/104772500/904027b0-5c6b-4a1e-bc12-65515dd1ed9a)

The screen has been updated and the Querying section is actually removed from the data source, so this information can be removed.

Author: Actually, it's not removed from the data source. It just wasn't present in the data source that I was looking at. .

Author: What if we minimize this information? How is this? 
@simonswine 

```suggestion
**Minimum step** is used for queries returning time-series data. The default value is 15 seconds.

Adjusting this option can help prevent gaps when you zoom in to profiling data. 
```

Reviewer 2: That's perfect

### Private data source connect

The **Private data source connect** section is shown in number 4 in the screenshot.

This option lets you query data that lives within a secured network without opening the network to inbound traffic from Grafana Cloud.

Use the drop-down box to select a configured private data sources.

> Select **Manage private data source connect** to configure and manage any private data sources you have configured.

Reviewer 2: Private data source connect is something that Grafana Cloud only. (correct me if I am wrong), so I don't think it should be mentioned here.

Reviewer 2: E.g. Prometheus also doesn't mention this: https://github.com/grafana/grafana/blob/a16782a2d6c5823dd8d1abdc463da9f8d8b8113a/docs/sources/datasources/prometheus/configure-prometheus-data-source.md?plain=1#L112
